src/_P020_Ser2Net.ino

```diff
@@ -288,6 +288,34 @@ boolean Plugin_020(uint8_t function, struct EventStruct *event, String& string)
         success = true;
       }
 
+      if (command == F("serialsendhex")){
+        success = true;
+        String tmpString = string.substring(14);
+        unsigned int len = tmpString.length();
+        const char* buf = tmpString.c_str();
+        for(unsigned int i=0; i < len / 2; i++){
+          uint8_t uc = buf[i * 2], dc = buf[i * 2 + 1];
+          if(uc >= 48 && uc <= 57) {
+              uc -= 48;
+          } else if(uc >= 65 && uc <= 70) {
+              uc -= 55;
+          } else if(uc >= 97 && uc <= 102) {
+              uc -= 87;
+          }
+
+          if(dc >= 48 && dc <= 57) {
+              dc -= 48;
+          } else if(dc >= 65 && dc <= 70) {
+              dc -= 55;
+          } else if(dc >= 97 && dc <= 102) {
+              dc -= 87;
+          }
+          task->ser2netSerial->write(uc << 4 | dc);
+        }
```

Comment:
I think this code may need some commenting, and/or can be simplified a lot.
It looks like you're trying to parse hex to binary.
Why not using `strtoul` ? See [cppreference - strtoul](https://en.cppreference.com/w/c/string/byte/strtoul)
Base is 16 , so it will probably be something like this: `strtoul(buf,buf+1,16)` Have to check if the endpointer is the last address or should be the address past where you're supposed to read.
The resulting value should then probably be `static_cast<uint8_t>(value)` before writing to the serial port.